pkg/api/types.go

```diff
@@ -1143,10 +1143,28 @@ type OperatorStepConfiguration struct {
 	Substitutions []PullSpecSubstitution `json:"substitutions,omitempty"`
 }
 
-// Bundle contains the data needed to build a bundle from the bundle source image
+// IndexUpdate specifies the update mode for an operator being added to an index
+type IndexUpdate string
+
+const (
+	IndexUpdateSemver          = "semver"
+	IndexUpdateReplaces        = "replaces"
+	IndexUpdateSemverSkippatch = "semver-skippatch"
+)
+
+// Bundle contains the data needed to build a bundle from the bundle source image and update an index to include the new bundle
 type Bundle struct {
+	// As defines the name for this bundle
+	As string `json:"as,omitempty"`
```

Comment:
I think `As` being optional (omitempty) may be a bit confusing, we may want to point out in the GoDoc that it's intentional.

Reply:
Added